DeepSpeech.py

```diff
@@ -670,12 +671,77 @@ def __call__(self, progress, data, **kwargs):
         log_info('FINISHED optimization in {}'.format(datetime.utcnow() - train_start_time))
     log_debug('Session closed.')
 
+def finetune_lm_params():
+    original_alpha, original_beta = FLAGS.lm_alpha, FLAGS.lm_beta
+    tmp_csv_path = os.path.join(FLAGS.finetune_temp_dir, '__TMP_FINETUNE_SAMPLE__.csv')
+    df = read_csvs(FLAGS.finetune_lm_csv_files.split(','))
+    n_sample = min(df.shape[0], FLAGS.finetune_lm_sampling_size)
+    df = df.sample(replace=False, n=n_sample)
+    df.to_csv(tmp_csv_path, index=None)
+
+    stats = []
+    for alpha in np.linspace(start=FLAGS.finetune_lm_alpha_min, stop=FLAGS.finetune_lm_alpha_max, num=FLAGS.finetune_lm_alpha_steps):
+        print('\t>>>> [FINETUNE LM] Test alpha = {} <<<<'.format(alpha))
+        FLAGS.lm_alpha = alpha
+        tfv1.reset_default_graph()
+        samples = evaluate([tmp_csv_path], create_model, try_loading)
+        for sample in samples:
+            stats.append({
+                'param': alpha,
+                'result': sample['word_distance'],
+            })
+
+    best_alpha = finetune_parabola(stats, FLAGS.finetune_lm_alpha_min, FLAGS.finetune_lm_alpha_max)
+    print('[FINETUNE LM] Set best alpha = {}'.format(best_alpha))
+    FLAGS.lm_alpha = best_alpha
+
+    stats = []
+    for beta in np.linspace(start=FLAGS.finetune_lm_beta_min, stop=FLAGS.finetune_lm_beta_max, num=FLAGS.finetune_lm_beta_steps):
+        print('\t>>>> [FINETUNE LM] Test beta = {} <<<<'.format(beta))
+        FLAGS.lm_beta = beta
```

Comment:
Instead of overwriting the global flag, refactor the called functions to be able to take these values as parameters and pass them in.